mne/forward/tests/test_field_interpolation.py

```diff
@@ -154,4 +159,45 @@ def test_make_field_map_meg():
     assert_raises(ValueError, make_field_map, evoked, ch_type='foobar')
 
 
+def _setup_args(info):
+    coils = _create_coils(info['chs'], FIFF.FWD_COIL_ACCURACY_NORMAL,
+                          info['dev_head_t'], 'meg')
+    my_origin, int_rad, noise, lut_fun, n_fact = _setup_dots('fast',
+                                                             coils,
+                                                             'meg')
+    args_dict = dict(intrad=int_rad, volume=False, coils1=coils, r0=my_origin,
+                     ch_type='meg', lut=lut_fun, n_fact=n_fact)
+    return args_dict
+
+
+@testing.requires_testing_data
+def test_as_meg_type_evoked():
+    """Test interpolation of data on to virtual channels."""
+
+    # validation tests
+    evoked = read_evokeds(evoked_fname, condition='Left Auditory')
+    assert_raises(ValueError, evoked.as_type, 'meg')
+
+    # channel names
+    ch_names = evoked.info['ch_names']
+    virt_evoked = evoked.pick_channels(ch_names=ch_names[:10:1],
+                                       copy=True).as_type('mag')
+    assert_true(all(['_virtual' in ch for ch in virt_evoked.info['ch_names']]))
+
+    # pick from and to channels
+    evoked_from = evoked.pick_channels(ch_names=ch_names[2:10:3], copy=True)
+    evoked_to = evoked.pick_channels(ch_names=ch_names[0:10:3])
+
+    info_from, info_to = evoked_from.info, evoked_to.info
+
+    # set up things
+    args1, args2 = _setup_args(info_from), _setup_args(info_to)
+    args1.update(coils2=args2['coils1']), args2.update(coils2=args1['coils1'])
+
+    # test cross dots
+    cross_dots1 = _do_cross_dots(**args1)
+    cross_dots2 = _do_cross_dots(**args2)
+
+    assert_array_almost_equal(cross_dots1, cross_dots2.T)
```

Comment:
I think there is another test you can do, which is a more transparent and practical one -- basically the tests you were doing visually. Use mags to interpolate to grads, compare to real grads. Use grads to interpolate to mags, compare to real mags. `np.allclose` will almost certainly not work since the accuracy isn't that high -- I'd see if a `np.corrcoef` over the `.ravel()`'ed result was e.g. 0.99 or higher.